apps/snitch_core/lib/core/data/model/general_configuration.ex

```diff
@@ -10,6 +10,12 @@ defmodule Snitch.Data.Model.GeneralConfiguration do
   alias Snitch.Tools.Helper.ImageUploader
   alias Ecto.Multi
 
+  @spec fetch_currency() :: String.t()
+  def fetch_currency do
+    general_config = Repo.all(GC) |> List.first()
```

Comment:
Why `Repo.all()` when we can use `Repo.one()`? That way you also won't have to do `List.first()`

Reply:
yes, then I have to handle the scenario where multiple results for Repo.one() could be returned.

Reply:
`Repo.one` only returns just one result. In any case my assumption is that we can only have one general setting with the way how the code is written.